electrumx/server/session.py

```diff
@@ -1686,3 +1687,35 @@ def truncate_auxpow(self, headers_full_hex, start_height):
             height += 1
 
         return headers.hex()
+
+
+class BitcoinVaultElectrumX(ElectrumX):
+
+    async def get_balance(self, hashX):
+        utxos = await self.db.all_utxos(hashX)
+        confirmed = sum(utxo.value for utxo in utxos if utxo.spent_height == 0)
+        incoming = sum(utxo.value for utxo in utxos if utxo.spent_height == -1)
+        outgoing = sum(utxo.value for utxo in utxos if utxo.spent_height > 0)
+        unconfirmed = await self.mempool.balance_delta(hashX)
+        self.bump_cost(1.0 + len(utxos) / 50)
+        return {'confirmed': confirmed, 'unconfirmed': unconfirmed, 'incoming': incoming, 'outgoing': outgoing}
```

Comment:
We should use `alert_incoming` and `alert_outcoming`. Only `incoming` and `outgoing` might be misleading.

Reply:
Done